leap_year.rb

```diff
@@ -0,0 +1,18 @@
+def leap_year
+    leap_years = Array.new
+    print "Please enter a year to start. "
+    first_year = gets.chomp.to_i
+    print "Please enter a year to end. "
+    end_year = gets.chomp.to_i 
+  while first_year <= end_year
+    if first_year % 400 == 0
+      leap_years.push(first_year) 
+     elsif first_year % 100 == 0
+     elsif first_year % 4 == 0
+      leap_years.push(first_year) 
+      end
+      first_year += 1
```

Comment:
This variable name to me is a misnomer because it sounds like what the first input entered is, but that's only true during the first iteration of a loop. Defining a new integer variable is cheap, so it's preferable if it makes the program easier for a human to read, especially since a single integer is cheap in terms of performance.